New Package: Educator.Engine v1.0.5.15

lisa3907 opened the pull request:
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.10 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.10.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/304728)

lisa3907 committed:
New Package: Educator.Engine v1.0.5.15

lisa3907 commented:
## To support the domain validation for the installer URL, we have published TXT records under our publisher domain:

```
> nslookup -type=TXT _winget.educatorengine.com

_winget.educatorengine.com text = "owner=EducatorEngine"
_winget.educatorengine.com text = "s3-origin=jlsms-downloads.s3.ap-northeast-1.amazonaws.com"
```

This proves that the S3 bucket (jlsms-downloads.s3.ap-northeast-1.amazonaws.com) is officially controlled by the publisher (EducatorEngine).

## We have verified the installer URL for direct accessibility:

> curl -I "https://jlsms-downloads.s3.ap-northeast-1.amazonaws.com/Educator.Engine.Production-1.0.5.15-x64.msi"

```
HTTP/1.1 200 OK
x-amz-id-2: 2HQItboEpmT89ofD0nK/6zaFXctBmjxF8hHwpABYcOD0KvUTT1PW7OGYEGOOMptA7iOVu32I+AY=
x-amz-request-id: PP1S5SDMKDNJHB52
Date: Sun, 19 Oct 2025 04:38:49 GMT
Last-Modified: Sat, 18 Oct 2025 13:51:04 GMT
ETag: "5c20166d18055cd259576bf8fff9b00d-5"
x-amz-server-side-encryption: AES256
Accept-Ranges: bytes
Content-Type: binary/octet-stream
Content-Length: 85356544
Server: AmazonS3
```

This confirms the MSI is directly downloadable without redirects or authentication.


## Verified SHA256 checksum:

```powsershell
> Invoke-WebRequest "https://jlsms-downloads.s3.ap-northeast-1.amazonaws.com/Educator.Engine.Production-1.0.5.15-x64.msi" -OutFile .\Educator.Engine.Production-1.0.5.15-x64.msi
> Get-FileHash .\Educator.Engine.Production-1.0.5.15-x64.msi -Algorithm SHA256
Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          97754F1470359D7C4014AB3E1C854D2220B0FF2BCE5B462638B84E6DE2EC8274       D:\DEPLOY\JLSMS\Winget\Production\Educator.Engine.Production-1.0.5.15-x64.msi
```

```installer.ymal
InstallerSha256: 97754F1470359D7C4014AB3E1C854D2220B0FF2BCE5B462638B84E6DE2EC8274
```

The values match ✅


Kindly proceed with the manual review. Thank you!

lisa3907 commented:
At the moment, the installer is hosted on our public S3 bucket:

https://jlsms-downloads.s3.ap-northeast-1.amazonaws.com/Educator.Engine.Staging-1.0.5.15-x64.msi

This URL is publicly accessible, and we’ve verified its SHA256 checksum against the manifest.

We are in the process of updating our official site (https://educatorengine.com) to include a direct “Download” link that points to this installer, so that navigation from the PackageUrl to the InstallerUrl is straightforward.  
In the meantime, please use the above link as the publicly available reference.

Would this be sufficient for this PR, or should we wait until the website update is live before approval?

stephengillie reviewed (changes requested):
Hi @lisa3907,

> To support the domain validation for the installer URL, we have published TXT records under our publisher domain:

I'm not aware of a DNS validation feature for this product. Please do not risk compromizing your DNS (and security) to fulfill a validation requirement that hasn't been homologated. 

Manual Validation ended with:
<img width="556" height="359" alt="image" src="https://github.com/user-attachments/assets/65c97814-23ba-4d0a-8321-096036b9ce33" />

lisa3907 committed:
Update manifests/e/Educator/Engine/1.0.5.15/Educator.Engine.installer.yaml

Co-authored-by: Stephen Gillie <114952565+stephengillie@users.noreply.github.com>

lisa3907 commented:
Hi @stephengillie,

Thanks for the clear feedback.

1. **DNS TXT validation**  
Acknowledged — we understand that winget does not use DNS-based validation, and we’ve removed our earlier note about TXT records. We won’t rely on that mechanism.

2. **Dependencies**  
We’ve updated the manifest to include the .NET Desktop Runtime dependency as suggested:

```yaml
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.DotNet.DesktopRuntime.9
```
    
3. PackageUrl → InstallerUrl navigation
We are adding a public download page on our official site (e.g., https://educatorengine.com/download) that links directly to the MSI. Once live, we will update this PR with the URL.

Please let us know if any further adjustments are needed. Thank you!

lisa3907 committed:
Update PackageUrl to official download page  - Changed PackageUrl to https://www.educatorengine.com/html/sub01/sub0101.asp

lisa3907 commented:
Updated PackageUrl as requested.

lisa3907 committed:
Change InstallerSha256 in Educator.Engine installer

Updated the InstallerSha256 for the Educator.Engine package.

lisa3907 committed:
Merge branch 'microsoft:master' into Educator.Engine-1.0.5.15-de80e651-1466-48bc-b827-d6527a384acd
